lib/utils/escape-filter-value.js

```diff
@@ -27,22 +27,22 @@ function escapeFilterValue (toEscape) {
 }
 
 function escapeBuffer (buf) {
-  let result = ''
+  let bufLen = buf.length;
+  let result = '';
   for (let i = 0; i < buf.length; i += 1) {
     if (buf[i] >= 0xc0 && buf[i] <= 0xdf) {
       // Represents the first byte in a 2-byte UTF-8 character.
-      result += '\\' + buf[i].toString(16) + '\\' + buf[i + 1].toString(16)
+      result += '\\' + buf[i].toString(16);
+      if(i<bufLen-1) { result += '\\' + buf[i + 1].toString(16); }  //Only add second byte if exist in buffer
```

Comment:
This is incomplete (the three byte condition is also incomplete):

```suggestion
      if (i < bufLen-1) {
        result += '\\' + buf[i + 1].toString(16)
        i += 2
        continue
      }
```

Reply:
Hello, Why it's incomplete ?
Also i increment and continu does not need to be in the if statment because the if is not executed only at end of the buffer so there is no more iteration in the for loop.

Also, I confirm that this code works perfectly.